src/ZEO/ClientStorage.py

```diff
@@ -1388,6 +1389,14 @@ def verify_cache(self, server):
         for oid, tid in self._cache.contents():
             server.verify(oid, tid)
         server.endZeoVerify()
+
+        if server_tid > self._cache.getLastTid():
+            # We verified the cache, and got no new invalidations
+            # while doing so.  The records in the cache are valid,
+            # in that invalid current records were invalidated,
+            # but the last tid is wrong.  Let's fix it:
+            self._cache.setLastTid(server_tid)
```

Comment:
verify_cache is called at startup? 

Is there the possibility of invalidations being received while this method is running?  I'm wondering about race conditions possibly leading to setting the cache's lastTid *backwards*.